src/hooks/useAnalytics.ts

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { AnalyticsContext } from '../contexts/AnalyticsContext';
+import { AnalyticsUrlParams, createClient } from 'analytics-client';
+
+export const useAnalytics = () => {
+	const analytics = React.useContext(AnalyticsContext);
+	const { analyticsClient, urlParamsHandler, newQuery } = React.useMemo(() => {
+		if (
+			!analytics?.componentName ||
+			!analytics?.endpoint ||
+			!analytics?.projectName
+		) {
+			return {};
+		}
+		const urlParamsHandler = new AnalyticsUrlParams();
+		const newQuery =
+			urlParamsHandler.consumeUrlParameters(window.location.search) ?? null;
+		const passedDeviceId = urlParamsHandler.getPassedDeviceId();
```

Comment:
Maybe, instead of having the deviceId param defined here, the analytics context could have an extra optional callback prop for retrieving the extra props to pass to the `createClient` call.
Eg:
```ts
export interface Analytics {
	endpoint: string;
	projectName: string;
	componentName: string;
	componentVersion?: string;
	initializationExtra?: (urlParamsHandler: AnalyticsUrlParams) => AnyObject
}
```
where we would then provide something like:
```
initializationParameterTransformer: (urlParamsHandler) => ({
    passedDeviceId: urlParamsHandler.getPassedDeviceId(),
})
```